app/models/location.rb

```diff
@@ -127,10 +127,8 @@ class Location < AbstractModel
   versioned_class.before_save do |ver|
     ver.user_id = User.current_id || User.admin_id
     if (ver.version != 1) &&
-       Location.connection.select_value(%(
-         SELECT COUNT(*) FROM locations_versions
-         WHERE location_id = #{ver.location_id} AND user_id = #{ver.user_id}
-       )).to_s == "0"
+       Location::Version.where(location_id: ver.location_id,
+                               user_id: ver.user_id).none?
```

Comment:
If you prefer, `count` is more efficient as it fetches only the count from the db.
So the following (or however you prefer to format it).
```suggestion
       Location::Version.where(
         location_id: ver.location_id, user_id: ver.user_id
       ).count.zero?
```